### PR TITLE
langchain[minor]: Add stuff docs runnable

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/langchain/langchain/chains/combine_documents/stuff.py [hunk 1]
```diff
@@ -60,7 +126,7 @@ class StuffDocumentsChain(BaseCombineDocumentsChain):
     """LLM chain which is called with the formatted document string,
     along with any other inputs."""
     document_prompt: BasePromptTemplate = Field(
-        default_factory=_get_default_document_prompt
+        default_factory=lambda: DEFAULT_DOCUMENT_PROMPT
```
[the remark this answers is not in the document]
Comment (baskaryan) [reply libs/langchain/langchain/chains/combine_documents/stuff.py:129]: think its supposed to not need any input (see old _get_default_document_prompt)

File: libs/langchain/langchain/chains/combine_documents/stuff.py [hunk 2]
```diff
@@ -1,21 +1,87 @@
...
+    _document_prompt = document_prompt or DEFAULT_DOCUMENT_PROMPT
+    _output_parser = output_parser or StrOutputParser()
+
+    def _format_inputs(inputs: dict) -> dict:
+        inputs[DOCUMENTS_KEY] = document_separator.join(
+            format_document(doc, _document_prompt) for doc in inputs[DOCUMENTS_KEY]
+        )
+        return {k: v for k, v in inputs.items() if k in prompt.input_variables}
+
+    return _format_inputs | prompt | llm | _output_parser
```
[the remark this answers is not in the document]
Comment (baskaryan) [reply libs/langchain/langchain/chains/combine_documents/stuff.py:84]: _format_inputs is more readable, but also makes it so extra inputs are ignored which is actually quite nice when using stuff_docs in longer chains (which will happen)

File: libs/langchain/langchain/chains/combine_documents/stuff.py [hunk 3]
```diff
@@ -1,21 +1,93 @@
...
+    _validate_prompt(prompt)
+    _document_prompt = document_prompt or DEFAULT_DOCUMENT_PROMPT
+    _output_parser = output_parser or StrOutputParser()
+
+    def format_docs(inputs: dict) -> str:
+        return document_separator.join(
+            format_document(doc, _document_prompt) for doc in inputs[DOCUMENTS_KEY]
+        )
+
+    return (
```
Comment (hwchase17) [inline libs/langchain/langchain/chains/combine_documents/stuff.py:83]: does the DOCUMENT_KEY variable really save anything over `context=format_docs`?

File: libs/langchain/langchain/chains/combine_documents/base.py [hunk 4]
```diff
@@ -14,6 +18,21 @@
 from langchain.chains.base import Chain
 from langchain.text_splitter import RecursiveCharacterTextSplitter, TextSplitter
 
+LanguageModelLike = Union[
```
Comment (hwchase17) [inline libs/langchain/langchain/chains/combine_documents/base.py:21]: this should probably go in core, https://github.com/langchain-ai/langchain/pull/15084/files#diff-639e6297e1db3729c228f5172ad900f6b5385c04ffef072fba2c4acd76de7e8d has it there

File: libs/langchain/langchain/chains/combine_documents/stuff.py (hunk 3 above)
Comment (baskaryan) [reply libs/langchain/langchain/chains/combine_documents/stuff.py:83]: if we ever change DOCUMENT_KEY we'd forget to update context

**Pull request merged**